File: Atividade 1/1.2/tarefa.py
from sys import maxsize

numero
maior = 0
menor = maxsize
contadorNumeros = 0
NUMERO_DE_SAIDA = -1

while(True):
	contadorNumeros += 1
	
	#	Reinicia o valor do numero
	numero = NUMERO_DE_SAIDA - 1

	while(numero < NUMERO_DE_SAIDA):
		numero = int(input("Digite um numero: "))

	if(numero == NUMERO_DE_SAIDA):
		break

	if(numero > maior):
		maior = numero
	if(numero < menor):
		menor = numero

if contadorNumeros == 1:
	print("Nenhum valor valido foi digitado!")
else:
	print("O menor numero natural digitado foi " + str(menor))
	print("O maior numero natural digitado foi " + str(maior))

File: Atividade 1/1.2/Tarefa.java
import java.util.Scanner;

public class Tarefa {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int maior = 0;
		int menor = Integer.MAX_VALUE;
		int i = 0;
		final int NUMERO_DE_SAIDA = -1;
		
		while(true) {
			i++;
			
			int numerosDigitados;
			
			do{
				System.out.print("Digite o " + i + "-esimo numero natural: ");
				numerosDigitados = scan.nextInt();
			}while(numerosDigitados < NUMERO_DE_SAIDA);
			
			if(numerosDigitados == NUMERO_DE_SAIDA) break;
	
			maior = (numerosDigitados > maior) ? numerosDigitados : maior;
			menor = (numerosDigitados < menor) ? numerosDigitados : menor;
		}

		System.out.println("O menor numero natural digitado foi " + menor + "!");
		System.out.println("O maior numero natural digitado foi " + maior + "!");
		scan.close();
	}
}
